samlsp/samlsp.go

```diff
@@ -31,14 +31,26 @@ type Options struct {
 	CookieMaxAge      time.Duration
 	CookieSecure      bool
 	ForceAuthn        bool
+	MetadataPath      string
+	ACSPath           string
 }
 
 // New creates a new Middleware
 func New(opts Options) (*Middleware, error) {
 	metadataURL := opts.URL
-	metadataURL.Path = metadataURL.Path + "/saml/metadata"
+	if opts.MetadataPath == "" {
```

Comment:
this could be cleaned up with
`metadataURL.Path  +="/saml/metadata"

Reply:
fixed, thanks!